docs/source/common/trainer.rst

```diff
@@ -987,6 +987,23 @@ Trainer will train model for at least min_steps or min_epochs (latest).
     # Run at least for 100 steps (disable min_epochs)
     trainer = Trainer(min_steps=100, min_epochs=0)
 
+max_time
+^^^^^^^^
+
+Set the maximum amount of time for training. Training will get interrupted mid-epoch.
+For customizable options use the :class:`~pytorch_lightning.callbacks.timer.Timer` callback.
+
+.. testcode::
+
+    # Default (disabled)
+    trainer = Trainer(max_time=None)
+
+    # Stop after 12 hours of training or when reaching 10 epochs
+    trainer = Trainer(max_time="00:12:00:00", max_epochs=10)
```

Comment:
Now that we also support days, this has become more complex.

And you know people will want other inputs supported.

Should we just allow instead

```python
    trainer = Trainer(max_time={'hours': 12}, max_epochs=10)
```

Which is the `**kwargs` to timedelta?

Reply:
that's one option.
I was thinking maybe to support just the shortest format, i.e., "SS", "MM:SS", "HH:MM:SS", etc.

Reply:
If datetime offered something like `fromisoduration` similar to https://docs.python.org/3/library/datetime.html#datetime.date.fromisoformat then I'd like passing the iso string, but since that's not the case, I'd rather avoid us parsing the string into the timedelta

But no strong feelings about it

Reply:
okay, I agree doing the parsing ourself is not optimal. But if the max_time argument is exposed in the trainer, I think we need at least a string input for the default CLI experience.

Reply:
any way to get the CLI to work with timedelta? then we save ourselves this hassle